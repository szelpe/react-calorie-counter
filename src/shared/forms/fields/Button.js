import React from 'react';

function Button(props) {
    return (
        <div>
            <button type={props.type} className="btn btn-primary">{props.children}</button>
        </div>
    );
}

export default Button;