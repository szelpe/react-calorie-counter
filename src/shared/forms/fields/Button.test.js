import Button from './Button';
import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

beforeEach(()=>{});
afterEach(()=>{});

test('Button should render correctly', () => {
    let div = document.createElement('div');
    ReactDOM.render(<Button type="button">hello</Button>, div);

    //console.log(div);

    let buttonText = div.querySelector('button').textContent;

    expect(buttonText).toBe('hello');
});

test('Button should render correctly using mount', () => {

    let wrapper = mount(<Button type="button">hello</Button>);

    //console.log(wrapper.debug());

    expect(wrapper.find('button')).toHaveText('hello');
});

test('snapshot testing button', () => {
    let json = renderer
        .create(<Button type="button">hello</Button>)
        .toJSON();

    expect(json).toMatchSnapshot();
});
