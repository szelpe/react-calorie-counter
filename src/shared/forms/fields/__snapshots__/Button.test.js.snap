// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing button 1`] = `
<div>
  <button
    className="btn btn-primary"
    type="button"
  >
    hello
  </button>
</div>
`;
